### PR TITLE
Add "rails routes --expanded" mode

### PR DESCRIPTION
**Pull request by benoittgt**
### Summary

When using rails routes with small terminal or complicated routes it can be very difficult to understand where a specific element is, especially when using header. psql had the same issue, that's why they created "expanded mode" [long time ago](https://github.com/postgres/postgres/commit/a45195a191eec367a4f305bb71ab541d17a3b9f9#diff-c8870df945d6a1270e9d5ed30e41d3f4R107) that you can
switch using `\x` or by starting psql with
```
-x
--expanded

    Turn on the expanded table formatting mode. This is equivalent to the \x command.
```
The output is similar to one implemented here for `rails routes`:

```
db_user-# \du
List of roles
-[ RECORD 1 ]----------------------------------------------
Role name  | super
Attributes | Superuser, Create role, Create DB
Member of  | {}
-[ RECORD 2 ]----------------------------------------------
Role name  | role
Attributes | Superuser, Create role, Create DB, Replication
Member of  | {}
```

Here is an output expanded mode (I removed active storage routes):
```
$ rails routes --expanded
--[ Route 1 ]------------------------------------------------------------
Prefix            | high_scores
Verb              | GET
URI               | /high_scores(.:format)
Controller#Action | high_scores#index
--[ Route 2 ]------------------------------------------------------------
Prefix            | new_high_score
Verb              | GET
URI               | /high_scores/new(.:format)
Controller#Action | high_scores#new
--[ Route 3 ]------------------------------------------------------------
Prefix            | blog
Verb              |
URI               | /blog
Controller#Action | Blog::Engine

[ Routes for Blog::Engine ]
--[ Route 1 ]------------------------------------------------------------
Prefix            | cart
Verb              | GET
URI               | /cart(.:format)
Controller#Action | cart#show
````

### Other Information

- I didn't add an entry to changelog because it seems too early for me. Also, I didn't know where you want this to be targeted if it was approved.
- [App used to display routes](https://github.com/benoittgt/rails_app_test_rake_routes)

### CODE REVIEW

**Comment by jaredbeck** [comment]
Good idea Benoit, this will be helpful. It'll be harder to `grep` this format, but we can't have everything. :)

**Comment by benoittgt** [comment]
@jaredbeck you can still use '-c controller' or '-g pattern'

```
rails routes -g high --expanded
--[ Route 1 ]------------------------------------------------------------
Prefix            | high_scores
Verb              | GET
URI               | /high_scores(.:format)
Controller#Action | high_scores#index
--[ Route 2 ]------------------------------------------------------------
Prefix            | new_high_score
Verb              | GET
URI               | /high_scores/new(.:format)
Controller#Action | high_scores#new
```

**Comment by jaredbeck** [comment]
> @jaredbeck you can still use '-c controller' or '-g pattern'

Awesome. I had no idea about either of those, thanks.

**Review by rafaelfranca — COMMENTED** [review]
Looks good to me. Can you add a CHANGELOG entry to the railties changelog?

File: railties/lib/rails/commands/routes/routes_command.rb [hunk 1]
```diff
@@ -7,12 +7,14 @@ module Command
     class RoutesCommand < Base # :nodoc:
       class_option :controller, aliases: "-c", type: :string, desc: "Specifies the controller."
       class_option :grep_pattern, aliases: "-g", type: :string, desc: "Specifies grep pattern."
+      class_option :expanded_format, aliases: "--expanded", type: :string, desc: "Specifies grep pattern."
```
Comment (yskkin) [inline railties/lib/rails/commands/routes/routes_command.rb]: I believe that `desc` should be rewrited.

Comment (benoittgt) [reply railties/lib/rails/commands/routes/routes_command.rb]: ouppps correct

**Comment by benoittgt** [comment]
When it's good I can squash.

**Comment by benoittgt** [comment]
Squashed 🎉

**Comment by Schwad** [comment]
Really excited for this feature, thank you so much for your work. 🎆

**Pull request merged**